ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/ClasspathComputer.java

```diff
@@ -65,79 +72,86 @@ public static void setClasspath(IProject project, IPluginModelBase model) throws
 
 	public static IClasspathEntry[] getClasspath(IProject project, IPluginModelBase model, Map<String, IPath> sourceLibraryMap, boolean clear, boolean overrideCompliance) throws CoreException {
 		IJavaProject javaProject = JavaCore.create(project);
-		ArrayList<IClasspathEntry> result = new ArrayList<>();
-		IBuild build = getBuild(project);
+		List<IClasspathEntry> originalClasspath = clear ? List.of() : Arrays.asList(javaProject.getRawClasspath());
```

Comment:
Shouldn't we call `updateTestAttribute` or something similar on retained original source-entries if the plugin is a test-plugin?
Maybe in `collectInOriginalOrder()`?

Reply:
Isn't the TEST attribute preserved together with any other attribute we preserve anyway?

Reply:
They are (probably) preserved, but IIRC the old implementation also added them if the project had a Test-Name, but the source entries are not marked as test (e.g. for an old Project).

Reply:
I had a quick look and pushed another suggestion. Simply updating all src-entries in the end would ensure it is in all cases of old/retained and new entries and whose test-attribute aligns with the project having a test-name or not. And it would also avoid the need to pass the flag if the project has a test-name or not down all the methods.
What do you think?